client/src/components/TaskReflectionModal/TaskReflectionModal.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from "react";
+import styles from "./TaskReflectionModal.module.css";
+import { PropTypes } from "prop-types";
+
+const RatingDot = ({ chosen, onChosen }) => {
+  return (
+    <span
+      className={chosen ? styles.chosenDot : styles.ratingDot}
+      onClick={onChosen}
+    ></span>
+  );
+};
+const TaskReflectionModal = ({ onSubmit, onClose, onDone }) => {
+  // chosenDot is the index of the chosen dot (0-9)
+  const [chosenDot, setChosenDot] = useState(0);
```

Comment:
Minor comment: I would rename this to `selectedDot`

Reply:
Renamed.